### PR TITLE
Update README to include missing boards

### PR DESCRIPTION
**Pull request by JoshuaLowe1002**
I couldn't find Hallowing under the README so I thought it should be put in. Follows the same format as the other board (product page + circuitpython guide)

### CODE REVIEW

**Comment by JoshuaLowe1002** [comment]
Yes! Sure thing!

**Comment by JoshuaLowe1002** [comment]
**should** all be updated now with the boards. Also added M0 and M4 labels

File: README.rst [hunk 1]
```diff
@@ -44,7 +44,7 @@ Designed for CircuitPython
 
 -  `Adafruit Feather M4 Express <https://www.adafruit.com/product/3857>`__ (`CircuitPython Guide <https://learn.adafruit.com/adafruit-feather-m4-express-atsamd51/circuitpython>`__)
 -  `Adafruit ItsyBitsy M4 Express <https://www.adafruit.com/product/3800>`__ (`CircuitPython Guide <https://learn.adafruit.com/introducing-adafruit-itsybitsy-m4/circuitpython>`__)
--  `Adafruit Metro M4 <https://www.adafruit.com/product/3382>`__ (`CircuitPython Guide <https://learn.adafruit.com/adafruit-metro-m4-express-featuring-atsamd51/overview>`__)
+-  `Adafruit Metro M4 <https://www.adafruit.com/product/3382>`__ (`CircuitPython Guide <https://learn.adafruit.com/adafruit-metro-m4-express-featuring-atsamd51/circuitpython>`__)
```
Comment (dhalbert) [inline README.rst]: Could you change this to "Metro M4 Express"? That's the last thing. Thanks! The product pages don't say Express, but the guide pages do, and it is an Express board.

Comment (JoshuaLowe1002) [reply README.rst]: Whoops I forgot that one, Done!

**Comment by JoshuaLowe1002** [comment]
Whoops I forgot that one, Done!

**Comment by sabas1080** [comment]
Add Meow Meow please

**Comment by dhalbert** [comment]
@sabas1080 We are just listing Adafruit products in the README.

**Comment by tannewt** [comment]
We should list non-Adafruit boards as well. It can be a separate list though so that its clear its not made by Adafruit.

**Comment by dhalbert** [comment]
@sabas1080 @deshipu Your boards are in the `boards/` directory. Feel free to start a new list "third-party" or "non-Adafruit" boards list with names and links to your things.

**Pull request merged**